app/controllers/posts_controller.rb

```diff
@@ -8,14 +8,25 @@ def create
         if @post.save
             redirect_to root_path
         else
-            render 'end'
+            @errors = @post.errors
         end 
     end
 
     def show
         @post = Post.find(params[:id])
     end
 
+    def destroy
+        @post = Post.find(params[:id])
+        if @post.destroy
+            flash[:notice] ="Post deleted"
+            redirect_to root_path
+        else
+            flash[:notice]="Failed to delete post"
+            render
+        end
```

Comment:
Rails flash mechanism comes with the options of `:notice` (blue), `:alert` (yellow), `:error` (red).
You can use either but it should change the color of the flash message to better suit the context.

So for a failed delete, you might find `:error` is more helpful to the user but its not a necessity